src/Agent.Sdk/ITraceWriter.cs

```diff
@@ -1,11 +1,18 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+using System;
 
 namespace Agent.Sdk
 {
     public interface ITraceWriter
```

Comment:
The relationship  between `IBaseContext` and `ITraceWriter` still feels a bit weird? I think the implementation of `ITraceWriter` will always be the same, right? If so it might be good to have some default implementation.

Reply:
it is weird, but I am not sure I understand your question.